src/xhk.c

```diff
@@ -349,7 +352,10 @@ int ProcessKeycode(XWindowsScreen_t * screen, int keycode, int up_flag)
         space = SPACE_STATE_MODIFIED; /* Space bar can no longer insert a space char */
         keycode = mirrored_key;
     }
-
+    if(keycode == KEY_ESC && space != SPACE_STATE_START) {
+        space = SPACE_STATE_MODIFIED;
```

Comment:
Should this be going to STATE_MODIFIED? I thought your commit messages suggested that 'escape' should 'cancel' the inversion ?

Reply:
Also - I prefer the 'one commit - one thing' approach - so it would be better if the escape handling was in it's own patch.

Reply:
STATE_MODIFIED is intended, to prevent any character  population after KEY_SPACE released.
It looks comment to commit is little bit ambiguous.  The goal was make it possible cancel unintentionally pressed key modifier(space), instead of use additional key sequence to remove resulting space.

Reply:
> Also - I prefer the 'one commit - one thing' approach - so it would be better if the escape handling > was in it's own patch.

I've spited changes on two independent pull requests (haven't found simple way to rebase history at current PR):
- [esc to cancel modifier](https://github.com/kbingham/xhk/pull/6)
- [grave/minus mapping](https://github.com/kbingham/xhk/pull/5)

Let's cancel current PR (#4)